tests/test_coordinate_source.py

```diff
@@ -0,0 +1,328 @@
+            shutil.copyfile(self.original_mrc_path, _new_mrc_path)
+
+            # get the first half of the coordinates for i=0 [0:220] and the
+            # second half for i=1 [220:440]
+            _centers = centers[i * 220 : (i + 1) * 220]
+
+            # create a coord file (only particle centers listed)
+            self.coord_fp = os.path.join(self.data_folder, f"sample{i+1}.coord")
+            # create a star file (only particle centers listed)
+            self.star_fp = os.path.join(self.data_folder, f"sample{i+1}.star")
+            # create a box file (lower left corner and X/Y sizes)
+            self.box_fp = os.path.join(self.data_folder, f"sample{i+1}.box")
+            # box file with nonsquare particles
+            self.box_fp_nonsquare = os.path.join(
+                self.data_folder, f"nonsquare_sample{i+1}.box"
+            )
+
+            # populate coord file with particle centers
+            with open(self.coord_fp, "w") as coord:
+                for center in _centers:
+                    # .coord file usually contains just the centers
+                    coord.write(f"{center[0]}\t{center[1]}\n")
+
+            # populate star file with particle centers
+            x_coords = [center[0] for center in _centers]
+            y_coords = [center[1] for center in _centers]
+            blocks = OrderedDict(
+                {
+                    "coordinates": DataFrame(
+                        {"_rlnCoordinateX": x_coords, "_rlnCoordinateY": y_coords}
+                    )
+                }
+            )
+            starfile = StarFile(blocks=blocks)
+            starfile.write(self.star_fp)
+
+            # populate box file with coordinates in EMAN1 .box format
+            with open(self.box_fp, "w") as box:
+                for center in _centers:
+                    # to make a box file, we convert the centers to lower left
+                    # corners by subtracting half the particle size (here: 256)
+                    lower_left_corners = (center[0] - 128, center[1] - 128)
+                    box.write(
+                        f"{lower_left_corners[0]}\t{lower_left_corners[1]}\t256\t256\n"
+                    )
+            # populate the second box file with nonsquare coordinates
+            with open(self.box_fp_nonsquare, "w") as box_nonsquare:
+                for center in _centers:
+                    # make a bad box file with non square particles
+                    lower_left_corners = (center[0] - 128, center[1] - 128)
+                    box_nonsquare.write(
+                        f"{lower_left_corners[0]}\t{lower_left_corners[1]}\t256\t100\n"
+                    )
+
+        # Now construct a simulated Relion directory
+        # to test RelionCoordinateSource
+        # The autopick.star file points to sample1.mrc
+        # (created earlier) and relion_coord.star as the coord file
+        os.makedirs(os.path.join(self.data_folder, "AutoPick/job006/"))
```

Comment:
Move one line below and change to `os.makedirs(os.path.dirname(self.autopick_star_path))`.